spring-batch-bigquery/pom.xml

```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2006-2021 the original author or authors.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~       https://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.springframework.batch</groupId>
+    <artifactId>spring-batch-bigquery</artifactId>
+    <version>0.1.0-SNAPSHOT</version>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <java.version>15</java.version>
```

Comment:
Is there any dependency requiring Java 15? Otherwise this should be Java 8.

Reply:
JDK10+ API I used:

- java.util.function.Predicate#not
- java.util.stream.Collectors#toUnmodifiableList

If we have to support JDK8 environment, why cannot we just compile code with JDK 8 level but write with latest Java version (at least LTS)?

Reply:
Yes, we have to support the same base line as Spring Batch, which is currently Java 8 for Spring Batch v4. This means I should be able to import this extension in a Spring Batch 4 based project and be able to run it with Java 8. If I understand correctly, this is not possible since you are using APIs from Java 10+.

Reply:
Got it. Will update code to be compatible with JDK8.
By the way, is there any plan update Java version for Spring Batch core?

Reply:
Spring Batch v5 will be based on Spring Framework v6, so the minimum Java version will depend on the one required by Spring Framework (which has not been decided yet at the time of writing this).

Reply:
Done